File: packages/pixlive/android/src/main/java/io/capawesome/capacitorjs/plugins/pixlive/classes/options/ResizeARViewOptions.java
package io.capawesome.capacitorjs.plugins.pixlive.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.pixlive.classes.CustomExceptions;

public class ResizeARViewOptions {

    private final int x;
    private final int y;
    private final int width;
    private final int height;

    public ResizeARViewOptions(@NonNull PluginCall call) throws Exception {
        this.x = ResizeARViewOptions.getXFromCall(call);
        this.y = ResizeARViewOptions.getYFromCall(call);
        this.width = ResizeARViewOptions.getWidthFromCall(call);
        this.height = ResizeARViewOptions.getHeightFromCall(call);
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    private static int getXFromCall(@NonNull PluginCall call) throws Exception {
        Integer x = call.getInt("x");
        if (x == null) {
            throw CustomExceptions.X_MISSING;
        }
        return x;
    }

    private static int getYFromCall(@NonNull PluginCall call) throws Exception {
        Integer y = call.getInt("y");
        if (y == null) {
            throw CustomExceptions.Y_MISSING;
        }
        return y;
    }

    private static int getWidthFromCall(@NonNull PluginCall call) throws Exception {
        Integer width = call.getInt("width");
        if (width == null) {
            throw CustomExceptions.WIDTH_MISSING;
        }
        return width;
    }

    private static int getHeightFromCall(@NonNull PluginCall call) throws Exception {
        Integer height = call.getInt("height");
        if (height == null) {
            throw CustomExceptions.HEIGHT_MISSING;
        }
        return height;
    }
}

File: packages/pixlive/ios/Plugin/Classes/Options/ResizeARViewOptions.swift
import Capacitor

@objc public class ResizeARViewOptions: NSObject {
    // swiftlint:disable:next identifier_name
    let x: Int
    // swiftlint:disable:next identifier_name
    let y: Int
    let width: Int
    let height: Int

    init(_ call: CAPPluginCall) throws {
        self.x = try ResizeARViewOptions.getXFromCall(call)
        self.y = try ResizeARViewOptions.getYFromCall(call)
        self.width = try ResizeARViewOptions.getWidthFromCall(call)
        self.height = try ResizeARViewOptions.getHeightFromCall(call)
    }

    private static func getXFromCall(_ call: CAPPluginCall) throws -> Int {
        // swiftlint:disable:next identifier_name
        guard let x = call.getInt("x") else {
            throw CustomError.xMissing
        }
        return x
    }

    private static func getYFromCall(_ call: CAPPluginCall) throws -> Int {
        // swiftlint:disable:next identifier_name
        guard let y = call.getInt("y") else {
            throw CustomError.yMissing
        }
        return y
    }

    private static func getWidthFromCall(_ call: CAPPluginCall) throws -> Int {
        guard let width = call.getInt("width") else {
            throw CustomError.widthMissing
        }
        return width
    }

    private static func getHeightFromCall(_ call: CAPPluginCall) throws -> Int {
        guard let height = call.getInt("height") else {
            throw CustomError.heightMissing
        }
        return height
    }
}
